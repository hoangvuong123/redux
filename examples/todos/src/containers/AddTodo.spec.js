import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddTodo from './AddTodo';
import configureStore from 'redux-mock-store';

const setup = (setupProps = {}) => {
  const store = configureStore()({ todos: [] });
  const wrapper = shallow(<AddTodo store={store}/>);

  return { store, wrapper };
};

describe('AddTodo component', () => {
  it('render AddTodo component', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('handle Submit', () => {
    const { wrapper, store } = setup();
    const fakeEvent = { preventDefault: () => 'preventDefault' };
    const addTodoComponent = wrapper.dive();
    const instance = addTodoComponent.instance();
    instance.state.value = 'Add Todo'

    expect(addTodoComponent.find('form').length).toBe(1);
    addTodoComponent.find('form').simulate('submit', fakeEvent);
    expect(store.getActions()).toEqual([
      { id: 0, text: 'Add Todo', type: 'ADD_TODO' }
    ]);
  })

})
