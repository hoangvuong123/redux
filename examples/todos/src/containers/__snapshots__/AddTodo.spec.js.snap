// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo component render AddTodo component 1`] = `
<AddTodo
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
